src/pages/api/profile/get-profile.ts

```diff
@@ -32,62 +41,62 @@ const handler: Handler = async (request, response) => {
-      },
-      addresses: [
-        {
-          id: "0087969e-9a0e-4e24-b38c-45a7e89f2bc1",
-          sub: "edc29cc7-35d5-4409-850a-dab5e19beb58",
-          street_address: "905 Macejkovic Prairie",
-          locality: "Kautzerbury",
-          region: "Massachusetts",
-          postal_code: "76369-4333",
-          country: "Gambia",
-          kind: "nemo",
-          created_at: "2021-04-15T07:35:28.614Z",
-          updated_at: "2021-04-15T07:35:28.614Z",
-          street_address_2: "Suite 309",
-          primary: false,
-        },
-        {
-          id: "3bb53df5-3cfb-480b-8667-7e0f092c6ac1",
-          sub: "edc29cc7-35d5-4409-850a-dab5e19beb58",
-          street_address: "46974 Price Drives",
-          locality: "Port Jasminshire",
-          region: "Washington",
-          postal_code: "35501-1717",
-          country: "Western Sahara",
-          kind: "labore",
-          created_at: "2021-04-15T07:35:28.599Z",
-          updated_at: "2021-04-15T07:35:28.599Z",
-          street_address_2: "Apt. 098",
-          primary: true,
-        },
-      ],
-    };
-
-    span.setDisclosedAttribute("is user info fetched", true);
+
+    const profileAccessToken = await getProfileAccessToken(
+      userCookie.access_token,
+    );
+
+    span.setDisclosedAttribute("is profile access token available", true);
+
+    const profileClient = initProfileClient(profileAccessToken);
+
+    const { data: profileUserInfo } = await profileClient
+      .getProfile()
+      .catch((error) => {
+        span.setDisclosedAttribute("is profile user info fetched", false);
+
+        if (error.response.status === HttpStatus.UNAUTHORIZED) {
+          throw webErrorFactory(webErrors.invalidProfileToken);
+        }
+
+        if (error.response.status === HttpStatus.FORBIDDEN) {
+          throw webErrorFactory(webErrors.invalidScopes);
+        }
+
+        if (error.response.status === HttpStatus.NOT_FOUND) {
+          throw webErrorFactory(webErrors.profileUserNotFound);
+        }
+
+        throw webErrorFactory(webErrors.unreachable);
+      });
+
+    span.setDisclosedAttribute("is profile user info fetched", true);
+
+    const { data: profileAddresses } = await profileClient
+      .getAddresses()
+      .catch((error) => {
+        span.setDisclosedAttribute("is profile addresses fetched", false);
+
+        if (error.response.status === HttpStatus.UNAUTHORIZED) {
+          throw webErrorFactory(webErrors.invalidProfileToken);
+        }
+
+        if (error.response.status === HttpStatus.FORBIDDEN) {
+          throw webErrorFactory(webErrors.invalidScopes);
+        }
+
+        throw webErrorFactory(webErrors.unreachable);
+      });
+
+    span.setDisclosedAttribute("is profile addresses fetched", true);
+
     response.statusCode = HttpStatus.OK;
-    response.json({ userInfo: mockedUserInfo });
+    response.json({
+      profileData: {
+        profileUserInfo,
+        profileAddresses,
+      },
+    });
```

Comment:
Maybe we should go with
```ts
response.json({
    profileUserInfo,
    profileAddresses,
  });
```

What do you think?

Reply:
I agree